clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -4707,6 +4711,35 @@ void SYCLIntegrationHeader::emit(raw_ostream &O) {
     O << "  __SYCL_DLL_LOCAL\n";
     O << "  static constexpr bool isESIMD() { return " << K.IsESIMDKernel
       << "; }\n";
+    O << "  __SYCL_DLL_LOCAL\n";
+    O << "  static constexpr const char* getFileName() { return "
+#ifndef NDEBUG
+      << std::string(PLoc.getFilename())
+             .substr(std::string(PLoc.getFilename()).find_last_of("/\\") + 1)
```

Comment:
I trust that this works, but it might help if we have examples of a couple different ways the driver could send the filename information here for testing.   @mdtoguchi, can you help with different corner cases?

Reply:
The driver will pass whatever the user supplies on the command line.  It can be just the file name `file.cpp` or a full absolute name `/my/dir/file.cpp` or even just a relative path `../dir2/file.cpp`.

Reply:
I verified that this works for all the scenarios mentioned here, for both windows and linux, i.e., a total of six cases. 
@mdtoguchi @premanandrao are you suggesting specific driver tests for this? Else, I'll mark this as resolved.

Reply:
I don't think driver specific tests are needed here, but if it makes sense to verify with some -cc1 cases it should be considered.

Reply:
https://github.com/intel/llvm/pull/5335#issuecomment-1028323634